saphir_cli/src/openapi/generate/mod.rs

```diff
@@ -568,7 +573,7 @@ by using the --package flag."
         for field in &s.fields {
             if let Some(field_name) = field.ident.as_ref().and_then(|i| get_serde_field(i.to_string(), &field.attrs, &s.attrs)) {
                 if let Some(field_type_info) = TypeInfo::new(item.scope, &field.ty) {
-                    let field_as_ref = self.args.schema_granularity == SchemaGranularity::All && field_type_info.is_type_serializable;
+                    let field_as_ref = self.args.schema_granularity == SchemaGranularity::All && as_ref;
```

Comment:
This was a bug, preventing generation of referenced ("$ref") component when the struct was not implementing Serialize (which is perfectly valid for example in cases like a json request body, which only need to be Deserialize)